### PR TITLE
[NeoML] CTransformerEncoderLayer with pre normalization

### PR DESCRIPTION
**Pull request by favorart**

### CODE REVIEW

File: NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h [hunk 1]
```diff
@@ -55,6 +55,8 @@ class CActivationDesc;
 //      |          |     |
 //      +----------+     |
 //           |           |
+//   (attention PreNorm) |  // if preNorm set
```
Comment (masepi) [inline NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h]: It's better to draw two separate pictures.

File: NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h [hunk 2]
```diff
@@ -94,7 +100,7 @@ class NEOML_API CTransformerEncoderLayer : public CCompositeLayer {
 	int GetHiddenSize() const { return selfAttention->GetHiddenSize(); }
 	void SetHiddenSize( int hiddenSize );
 
-	// Dropout rate
+	// Dropout rate, probability of zero values, also set in attention layer
```
Comment (masepi) [inline NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h]: Why the same value?

File: NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h (hunk 1 above)
Comment (favorart) [reply NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h]: fixed

File: NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h (hunk 2 above)
Comment (favorart) [reply NeoML/include/NeoML/Dnn/Layers/TransformerLayer.h]: fixed

**Review by daniyalaliev — APPROVED** [review]

**Pull request merged**